runtime/shared_common/shrinit.cpp

```diff
@@ -3609,6 +3612,14 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 		returnVal = J9VMDLLMAIN_SILENT_EXIT_VM;
 	}
 
+	if (J9VMDLLMAIN_OK == returnVal) {
+		/* do not fetch Startup hints if returnVal is J9VMDLLMAIN_SILENT_EXIT_VM */
+		fetchStartupHintsFromSharedCache(currentThread);
```

Comment:
I'm thinking this should be called from j9shr_findGCHints() for now, to avoid doing anything until the gc hints are used.

Reply:
Fixed